db/controllers/articles.controller.js

```diff
@@ -1,10 +1,19 @@
-const { selectArticle } = require("../models/article.model");
+const { selectArticle, selectAllArticles } = require("../models/article.model");
 
-exports.getArticle = (req, res, next) => {
+exports.getArticles = (req, res, next) => {
   const { article_id } = req.params;
-  selectArticle(article_id)
-    .then((article) => {
-      res.status(200).send({ article });
-    })
-    .catch(next);
+  
+  if (!article_id) {
+    selectAllArticles()
```

Comment:
I feel like it may cause problems later on to have two different endpoints using the same controller - just something to be mindful of that you may need to separate this logic out into a getArticles and get**Article** controller at some point if you do run into issues but I could be wrong!

Reply:
I was originally planning to use the same model for both endpoint but decided to split it which I why I did it in the controller instead. Good point though and I'll definitely bear it in mind.